### PR TITLE
Fix python 3.7

### PR DESCRIPTION
**Pull request by gilesknap**
It's not that pretty but does pass tests in all python versions (and so does Malcolm4).

Perhaps when we drop Pythons < 3.7 we can review and simplify annotypes.

### CODE REVIEW

File: annotypes/_stackinfo.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# before its __orig_class__ member is set. This means there is
+# no way to verify that the correct types have been passed.
+#
+# As a workaround we walk up the stack to see which concrete
+# class called our __init__. This function does the stack walk
+# and is called from Array.__init__().
+
+
+def find_caller_class(filename):
+    print(filename)
```
Comment (thomascobb) [inline annotypes/_stackinfo.py]: Delete the print please

File: tests/test_annotypes.py [hunk 2]
```diff
@@ -162,6 +162,12 @@ def another_func(self):
         assert list(Sub1.call_types) == ["name"]
         assert list(Sub2.call_types) == ["name"]
 
+        # check subclassing, in particular that Any can be passed for
+        #  subclass checking
+        assert issubclass(Sub2, Root)
```
Comment (thomascobb) [inline tests/test_annotypes.py]: Not a fan of overriding a builtin. Can we make move the code from Malcolm that uses issubclass to a new classmethod Anno.matches_type(inst)?

Comment (gilesknap) [reply tests/test_annotypes.py]: I've added CallTypesMeta.matches_type and modified Malcolm Model to use it.

**Comment by gilesknap** [comment]
All suggestions completed.

**Pull request merged**